content/articles/creating-a-simple-chat-app-in-react-using-the-chat-engine-api/index.md

```diff
@@ -0,0 +1,231 @@
+2. Installed on your system Node.js, which you could access from [here](https://nodejs.org/en/download/). 
+3. [Node Package Manager (NPM)](https://docs.npmjs.com/cli/v6/commands/npm) know-how.
+
+### What Chat Engine API is?
+The above introduction gave us an overview of this API and we realized how the API could be consumed to build easily; chat services.
+It also allows the hosting of chats through the Rest API it lays out. The chat User Interface(UI), is created from the NPM components this tool comes with. See how the hustle is waved away for us already?
+Also, thanks to its server(s), the chat can be hosted from here(the server).
+
+
+### Creating the Application
+This minute, let's begin.
+- Head to your windows terminal and navigate to a new directory named to your preference, and create a  new react app. You could opt to use the Vite tool instead whose step-by-step guide is taught in [this](https://www.section.io/engineering-education/creating-a-react-app-using-vite/) article. 
+  Else, run the command below with the project name replaced with an application name of your choice. You may opt to use ***yarn create-react-app app-name***. 
+ 
+```bash
+npx create-react-app project-name
+```
+
+- In the created app folder, install the Chat Engine as a component using the command;
+
+```bash
+npm i react-chat-engine
+```
+
+#### Setting up a Chat Engine Project 
+- We are required to have the API keys which we will get after registering(signing up or logging in) at [chatengine.io](https://chatengine.io/) for an account.
+
+- You will be redirected to a page like below;
+  ![Project Setup](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/NewProject.png) 
+
+- Click the "New Project" button you see on the page and give your project a title.
+  By now, your browser shows such a page.
+  ![Project Keys](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/Keys.png) 
+  Save the Project ID and the secret keys somewhere safe. We will use them along the way.
+
+- Still, on this page, click the "new user" section at the top right corner and give a name to your new  user in a modal that pops up.
+  ![New User](/engineering-education/creating-a-simple-chat-app-in-react-the-using-chat-engine-api/user.png)
+
+> You may need to create a users' list to select from, when starting a direct chat later.
+
+As a developer, you may want to always practise the separation of **sensitive** information like your userSecret, you obtained and the projectID from the rest of the code to avoid such information being exposed to **unauthorized users**, hence protecting your data. 
+Therefore, create a .env file at the root of your directory and have the ***userSecret*** and ***projectID*** saved as environment variables.
+Remember to add the **.env** file to the **.gitignore** file to avoid pushing such sensitive data to Github.
```

Comment:
You have used `userSecret`, among others, kindly rewrite them well. such as ` user secret'.